tests/ssl-opt.sh

```diff
@@ -340,6 +341,23 @@ check_server_hello_time() {
     fi
 }
 
+# terminate server after giving it some time to wrap up and write logs
+terminate_server() {
+    # Shortcut the delay in case we know logs are ready.
+    # This only works for Mbed TLS, and not for all tests (sometimes with DTLS
+    # the server still doesn't know the client has disconnected at the end).
+    # However the shortcut test is designed to only have false negatives and
+    # no false positives, and we fall back to sleeping on negatives.
+    if tail -n-1 $SRV_OUT | grep -F 'Waiting for a remote connection' >/dev/null; then :;
```

Comment:
Technically this shortcut isn't actually correct: what if the server hasn't flushed its logs since it displayed the “Waiting” message?

Reply:
In principle, I agree with you. In practice, if the case you mentioned happened, then we would be seeing `server or client failed to reach handshake stage` errors (from around line 500) which I don't remember seeing. In general, I think the important point is that "shortcut + delay" is a strict improvement over the existing, ie all case that still fail after it also failed before.

However, I'll prepend `grep "Performing the SSL/TLS handshake" $SRV_OUT` to the shorcut condition for the sake of correcness.

Reply:
I'm ok with the current code. It isn't perfect, but it's good enough for what it does, which is to shorten a timeout that is itself imperfect.

I'm not sure if grepping “Performing the SSL/TLS handshake” is really better: there are a few tests that perform multiple connections to the server.

There's no perfect solution anyway. In theory, waiting for the server to flush its logs could take an arbitrary time, which is indistinguishable from a server that's stuck in an infinite loop. What matters is that it works in practice on CI systems, and if we aren't seeing truncated logs any longer then we're all good.